### ISSUE TITLE
[$1000] mWeb - Request money - There is No cursor in amount page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17714]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Click on the green plus button (FAB) -> Request Money
4. Enter an amount

## Expected Result:
Cursor is present

## Actual Result:
There is No cursor

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.2.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/233388714-96159742-38c4-4999-95fc-a000ada571a1.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b32376b47d699980</li>
        <li>Upwork Job ID: 1651248028633018368</li>
        <li>Last Price Increase: 2023-04-26</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @michaelhaxhiu (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@michaelhaxhiu Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by michaelhaxhiu** [comment]
Some talk about grouping this GH into a bigger auto-focus fix. I will review and update tomorrow

**Comment by michaelhaxhiu** [comment]
@sobitneupane do you think we should group this fix with others? It's a legit bug report - there should indeed be a cursor here.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01b32376b47d699980

**Comment by MelvinBot** [comment]
Current assignee @michaelhaxhiu is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by kuluruvineeth** [comment]
@michaelhaxhiu could you please add me to slack groups so that i can get instructions to setup for ios and android application for local development as of now i am able to run only web and desktop version, please help with the guidelines to setup for ios and android on mac m1, thanks.

**Comment by kuluruvineeth** [comment]
@jasperhuangg i figured out issues and now i am able to run on both ios and android, thanks.

**Comment by narefyev91** [comment]
Hi I'm Nicolay from Callstack - expert contributor group - will start investigation in this area

**Comment by MelvinBot** [comment]
📣 @narefyev91 You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by mountiny** [comment]
Assigning Nikolay to this as he has worked on this page previously, there are some consideration to have regarding the selection which is buggy on native and we are working on a bigger refactor of this. So maybe this could be on hold for that

**Comment by jasperhuangg** [comment]
Still waiting on @narefyev91's PR, not overdue!

**Comment by narefyev91** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Input component is not get focus when page is mounted

### What is the root cause of that problem?
Well i think there are a lot of discussion around this issue https://github.com/Expensify/App/issues/10414 - which is the same as we facing here.
But i would like to also paste my findings.
I'm not quite sure that we will be able to fix it because of:
1) Safari on mobile does not support autoFocus
<img width="671" alt="Screenshot 2023-05-04 at 10 14 26" src="https://user-images.githubusercontent.com/28352309/236186485-d76f1910-e182-43d7-9634-29fb30685985.png">
2) Safari prevent any manual on focus event - until user will make any interaction on the page https://bugs.webkit.org/show_bug.cgi?id=195884#c4

**From Apple team:**
We (Apple) like the current behaviour and do not want programmatic focus to bring up the keyboard when you do not have a hardware keyboard attached and the programmatic focus was not invoked in response to a user gesture. Why you may ask...because auto bringing up the software keyboard can be seen as annoying and a distraction to a user (not for your customers, but for everyone not using your app) given that:

1. We bring up the keyboard, which takes up valuable real estate on screen.
2. When we intent to bring up the software keyboard we zoom and scroll the page to give a pleasing input experience (or at least we hope it is pleasing; file bugs if not).

And i think even google was not able to make it working:
![photo_2023-05-04 14 08 33](https://user-images.githubusercontent.com/28352309/236187242-877b716b-f413-470b-93b4-ee157a9e39f4.jpeg)

### What changes do you think we should make in order to solve the problem?
The only potential fix here - which i think were already done before - when user click on numpad we add new value in input and focus a field.
```
    updateAmountNumberPad(key) {
        if (!this.textInput.isFocused()) {
            this.textInput.focus();
        }
```
The code already exists but it it not working as expected in safari ios - because when we mounted the page - this.textInput.isFocused() will be equal true - and that's why on safari ios - when you click on numpad - we see number - but not see cursor.
Possible solution here to create additional function same as above but with additional check something like this:
`if (!this.textInput.isFocused() || isMobileSafari()) focus input`

### What alternative solutions did you explore?

cc @mananjadhav @michaelhaxhiu @jasperhuangg

**Comment by mananjadhav** [comment]
Ohh I just saw the platform mentioned as iOS Safari, I thought it was native. I would recommend **:donothing:** here. This is the standard behavior by Apple, and I am sure that's the case across the app.

**Comment by MelvinBot** [comment]
@mananjadhav @michaelhaxhiu @narefyev91 @jasperhuangg this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by jasperhuangg** [comment]
Ah I agree, I think it's a relatively minor quirk that we shouldn't really be focusing our time on, especially if this is true:

> standard behavior by Apple, and I am sure that's the case across the app.

Closing, thanks everyone!
